labs/week 1 - r review and markdown intro.Rmd

```diff
@@ -0,0 +1,158 @@
+```{r load-gapminder}
+if(!require(gapminder))          # If gapminder package isn't intalled...
+  install.packages("gapminder")  # ... Install it
+
+minded <- gapminder::gapminder
+names(minded)
+
+```
+
+With that loaded up, let's do a little bit of data management. 
+```{r manage-gapminder}
+
+minded %>%   # Make sure you run the very first code chunk (lines 8-11) to load the tidyverse
+  mutate(
+    shortyear = year - 1950,             # Starting years at 1950
+    recent = case_when(
+      year > 2000 ~ "Post 2000", 
+      TRUE ~ "Pre 2000"                  # TRUE means "everything else" 
+      ),
+    newyears = case_when(
+      year %in% c(1997, 2002, 2007) ~ 1, # We can use this way to select recent years, too
+      TRUE ~ 0
+      )
+    ) %>% 
+  select(year, shortyear, recent, newyears)  # select(columns)
+
+newminder <- minded %>%            # Create new dataset 
+  rename(gdp_cap = gdpPercap) %>%  # rename(newname = oldname)
+  select(year, gdp_cap) %>% 
+  filter(gdp_cap < 1000)           # only include rows where gdp_gap is less than 1000
+newminder
+  
+
+```
+
+You can also see we have multiple outputs from the above code chunk. We can click at the top of the output pane that dropped down to go between them. 
+
+Also notice that, above each of these outputs, it shows the type of data (in this case tibbles) and the dimensions. 
+
+### Graphing with `ggplot2`
+
+```{r graph-gapminder}
+hist(minded$lifeExp)
+
+minded %>% 
+  ggplot(aes(x = lifeExp)) +
+  geom_histogram(bins = 25) +
+  labs(title = "Histograph of Life Expenctancy",
+       caption = "Data: Gapminder", 
+       x = "Life Expectancy (years)") 
+
+```
+
+Here is another example of a graph, this time with two variables. I also changed the color of the dots based on continents. 
+```{r graph-gapminder2}
+
+minded %>% 
+  ggplot(aes(x = year,
+             y = lifeExp)) +
+  geom_point(aes(color = continent, 
+                 size = pop)) + 
+  geom_smooth() +
+  theme_classic() +
+  labs(title = "Life Expectancy over Time",
+       x = "Year",
+       y = "Life Expectancy")
+  
+
+```
+
+
+## Working With Git and GitHub
+
+Git is a type of "version control" system. If you've ever saved a file and later wished you could go back to a previous version, Git is for you! 
+
+GitHub, then, is a place to store our code and all its previous versions. And like R, it's free.  
+
+We'll now walk through the process of installing Git and linking our Github to our RStudio project. The best resource for this is Jenny Bryan's Happy Git with R: https://happygitwithr.com/index.html 
+
+Chapters 4, 6, 7, 9-13, and 17 are the key sections here, and we'll walk through it together. 
+
+
```

Comment:
This all looks great. Perhaps as a demo they could do two things. 1. Make sure to clone the course repository and pull updates (we could update the repository after they have cloned it by adding a test file) and 2. Make their own Github repository and push an update. This might be too much for this session but these are the two things I want to emphasize: remembering to pull from the class repo and to push updates to their personal repos @ftraylor